Storage/FirebaseStorageUI/FUIStorageImageLoader.m

```diff
@@ -45,7 +51,8 @@ + (FUIStorageImageLoader *)sharedLoader {
 - (instancetype)init {
   self = [super init];
   if (self) {
-    self.defaultMaxImageSize = 10e6;
+    _defaultMaxImageSize = 10e6;
+    _coderQueue = dispatch_queue_create("com.google.firebaseui.storage.coderQueue", DISPATCH_QUEUE_SERIAL);
```

Comment:
Should we use a Loader (shared instance) level queue ?
Or, should we use per-download-task level queue ?

I'm afraid, if there are 10 concurrent download in the same time, this serial queue will been busy and block each download task.

Reply:
Implements the logic to use per-download-task level decoding queue.

This is the current suitable solution for this case, which can ensure callback order during progressive decoding (later appending data will neven produce image before previous one)